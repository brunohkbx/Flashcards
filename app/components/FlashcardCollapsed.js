import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import {
  Card,
  Title,
  Caption,
  Divider,
  Text,
  Button,
  Paragraph
} from 'react-native-paper';
import CenteredContainer from './CenteredContainer';

class FlashcardCollapsed extends Component {
  state = { collapsed: false }

  toggleCollapsedState = () => this.setState(prevState => {
    return { collapsed: !prevState.collapsed };
  })

  render() {
    const {
      currentQuestion,
      totalQuestions,
      flashcard
    } = this.props;

    const { collapsed } = this.state;

    return (
      <Card>
        <View style={{ padding: 8}}>
          <Text style={{ paddingTop: 10}}>
            <Title style={{fontSize: 30}}>{currentQuestion}</Title>
            <Caption style={{fontSize: 18}}> OF {totalQuestions}</Caption>
          </Text>
          <Divider />
        </View>
        <Card.Content>
          <CenteredContainer style={{ paddingVertical: 30}}>
            <Title>{flashcard.question}</Title>
            { collapsed && <Paragraph>{flashcard.answer}</Paragraph> }
          </CenteredContainer>
        </Card.Content>
        <Divider />
        <Card.Actions>
          <Button
            primary
            onPress={this.toggleCollapsedState}
          >
            { collapsed ? 'Hide Answer' : 'Show answer' }
          </Button>
        </Card.Actions>
      </Card>
    );
  }
}

FlashcardCollapsed.propTypes = {
  currentQuestion: PropTypes.number.isRequired,
  totalQuestions: PropTypes.number.isRequired,
  flashcard: PropTypes.shape({
    question: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired
  })
};

export default FlashcardCollapsed;
