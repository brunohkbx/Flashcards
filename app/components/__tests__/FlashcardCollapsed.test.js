import React from 'react';
import { shallow } from 'enzyme';
import FlashcardCollapsed from '../FlashcardCollapsed';

describe('FlashcardCollapsed', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      currentQuestion: 1,
      totalQuestions: 20,
      flashcard: {
        question: 'What is React?',
        answer: 'A library for managing user interfaces'
      }
    }, propOverrides);

    const wrapper = shallow(<FlashcardCollapsed {...defaultProps} />);
    const wrapperInstance = wrapper.instance();

    return { wrapper, wrapperInstance };
  }

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('displays answer when flashcard is not collapsed', () => {
    const { wrapper } = setup();

    wrapper.setState({ collapsed: true });

    expect(wrapper).toMatchSnapshot();
  });

  describe('toggleCollapsedState', () => {
    it('toggles the collapsed state', () => {
      const { wrapperInstance } = setup();
      wrapperInstance.setState({ collapsed: true });

      wrapperInstance.toggleCollapsedState();

      expect(wrapperInstance.state.collapsed).toBeFalsy();
    });
  });
});
