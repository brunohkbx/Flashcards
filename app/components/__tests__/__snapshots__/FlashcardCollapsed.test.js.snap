// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardCollapsed displays answer when flashcard is not collapsed 1`] = `
<withTheme(Card)>
  <Component
    style={
      Object {
        "padding": 8,
      }
    }
  >
    <withTheme(Text)
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <Title
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        1
      </Title>
      <Caption
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
         OF 
        20
      </Caption>
    </withTheme(Text)>
    <withTheme(Divider) />
  </Component>
  <Card.Content>
    <Styled(Component)
      style={
        Object {
          "paddingVertical": 30,
        }
      }
    >
      <Title>
        What is React?
      </Title>
      <Paragraph>
        A library for managing user interfaces
      </Paragraph>
    </Styled(Component)>
  </Card.Content>
  <withTheme(Divider) />
  <Card.Actions>
    <withTheme(Button)
      onPress={[Function]}
      primary={true}
    >
      Hide Answer
    </withTheme(Button)>
  </Card.Actions>
</withTheme(Card)>
`;

exports[`FlashcardCollapsed renders properly 1`] = `
<withTheme(Card)>
  <Component
    style={
      Object {
        "padding": 8,
      }
    }
  >
    <withTheme(Text)
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <Title
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        1
      </Title>
      <Caption
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
         OF 
        20
      </Caption>
    </withTheme(Text)>
    <withTheme(Divider) />
  </Component>
  <Card.Content>
    <Styled(Component)
      style={
        Object {
          "paddingVertical": 30,
        }
      }
    >
      <Title>
        What is React?
      </Title>
    </Styled(Component)>
  </Card.Content>
  <withTheme(Divider) />
  <Card.Actions>
    <withTheme(Button)
      onPress={[Function]}
      primary={true}
    >
      Show answer
    </withTheme(Button)>
  </Card.Actions>
</withTheme(Card)>
`;
